src/components/PopupWithForm.js

```diff
@@ -17,15 +18,27 @@ export default class PopupWithForm extends Popup {
     return this._inputValues;
   }
 
+  setButtonText(submit, buttonText = "Saving...") {
```

Comment:
You've implemented the code for rendering loading via the class methods. Great job!